app/main/views/drafts.py

```diff
@@ -180,37 +204,40 @@ def delete_draft_service(service_id):
     return jsonify(message="done"), 200
 
 
-@main.route('/services/<string:service_id>/draft/publish', methods=['POST'])
-def publish_draft_service(service_id):
+@main.route('/draft-services/<int:draft_id>/publish', methods=['POST'])
+def publish_draft_service(draft_id):
     """
     Publish a draft service
-    :param service_id:
+    :param draft_id:
     :return:
     """
 
-    is_valid_service_id_or_400(service_id)
-
     updater_json = validate_and_return_updater_request()
 
     draft = DraftService.query.filter(
-        DraftService.service_id == service_id
+        DraftService.id == draft_id
     ).first_or_404()
 
-    service = Service.query.filter(
-        Service.service_id == draft.service_id
-    ).first_or_404()
+    if draft.service_id:
+        service = Service.query.filter(
```

Comment:
This is somewhat similar to what the service update endpoint does, is it possible to extract it/parts of it to `service_utils`?